package/jest-utils/extractMockedSpecifier.js

```diff
@@ -0,0 +1,76 @@
+const babelTypes = require("@babel/types");
+
+function getMockedModuleProperties(node) {
+  const props = node.arguments?.[1]?.body?.properties;
+  if (props) return props;
+  return node.arguments?.[1]?.body.body?.[0]?.argument?.properties;
+}
+
+function setMockedModuleProperties(node, props) {
+  const isDirectReturn = node.arguments?.[1]?.body?.properties != null;
+
+  if (isDirectReturn) {
+    node.arguments[1].body.properties = props;
+  } else {
+    node.arguments[1].body.body[0].argument.properties = props;
+  }
+}
+
+function setDefaultReturn(node, returnNode) {
+  const isDirectReturn = node.arguments?.[1]?.body?.properties != null;
+
+  if (isDirectReturn) node.arguments[1].body = returnNode;
+  else node.arguments[1].body.body[0].argument = returnNode;
+}
+
+function extractMockedSpecifier(path, specifier, specifierOrigin, callback) {
+  const expression = path.parent;
+
+  const clone = babelTypes.cloneNode(expression);
+
+  let defaultOverride;
+
+  setMockedModuleProperties(
```

Comment:
This block to line 50 is the only part of your PR that I can't wrap my head around yet, would you be able to add a comment in the code explaining what is happening here?